packages/plugin-node-modules/sources/NodeModulesLinker.ts

```diff
@@ -636,36 +657,150 @@ const symlinkPromise = async (srcPath: PortablePath, dstPath: PortablePath) => {
   }
 };
 
-const copyPromise = async (dstDir: PortablePath, srcDir: PortablePath, {baseFs, innerLoop}: {baseFs: FakeFS<PortablePath>, innerLoop?: boolean}) => {
+async function checksumFile(path: PortablePath, baseFs: FakeFS<PortablePath>) {
+  return await new Promise<string>((resolve, reject) => {
+    const hash = crypto.createHash(`sha1`);
+    const stream = baseFs.createReadStream(path);
+
+    stream.on(`data`, chunk => {
+      hash.update(chunk);
+    });
+
+    stream.on(`error`, error => {
+      reject(error);
+    });
+
+    stream.on(`end`, () => {
+      resolve(hash.digest(`hex`));
+    });
+  });
+}
```

Comment:
Why not reuse the function from `hashUtils`?

Reply:
To utilize `hashUtils.checksumFile` I need more flexibility, I need to make it to use `baseFs` and not `xfs` all the time, plus I need to specify other hashing algorithm `SHA1` - for performance reasons. I have adjusted `hashUtils.checksumFile` to be more flexible and used it instead, please let me know if this approach looks better

Reply:
Is the perf impact sizable enough to warrant a different algorithm? We currently already use SHA256 everywhere even if *in theory* SHA1 would *likely* be enough, for consistency.

Regarding the baseFs thing, a few functions have an optional argument containing a bag of options with a `baseFs` property (for example [here](https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-core/sources/tgzUtils.ts#L26)) - it'd probably be best to just add that to the checksumFile function from the core.

Reply:
Yes, the perf impact is huge. With `sha1` install of `storybook` repo with hot storage takes `42.2s`, with `sha512` it takes `1m 55s`

> Regarding the baseFs thing, a few functions have an optional argument containing a bag of options with a baseFs property 

I have already did this in the PR